### ISSUE TITLE
Looking at a variable to see if NameError happens causes the detected value to become undefined

### ISSUE THREAD

**Issue by glyph** [pyjs/original-pyjs#787]
This bit of code

```
try:
    BaseException
except NameError:
    BaseException=Exception
print("BaseException is now: " + repr(BaseException))
```

prints

```
BaseException is now: undefined
```

but this disagrees with what Python prints.  (This sort of trick is useful for code that wishes to detect multiple Python versions and work regardless of its environment.)

**Comment by anthonyrisinger** [comment]
did you build with `--enable-strict`?  what was the build string?

**Comment by glyph** [comment]
I built with a bunch of different options (including `--enable-strict`); the output was always the same.

What do you mean by "build string"?  Can you not reproduce this trivially?

**Comment by anthonyrisinger** [comment]
nah, i wasn't in a place i could test/try, i simply wanted to ensure you were trying with strictness.  by build string i just meant the cmdline used to build.
